elasticsearch/helpers/actions.py

```diff
@@ -524,14 +524,27 @@ def scan(
                     resp["_shards"]["skipped"],
                     resp["_shards"]["total"],
                 )
+
+                resp_failures = resp["_shards"].get("failures")
+                failures = []
+                if resp_failures is not None:
+                    for resp_failure in resp_failures:
+                        failure = "Failure type[%s] received with reason: %s" % (
+                            resp_failure["reason"]["type"],
+                            resp_failure["reason"]["reason"],
+                        )
+                        failures.append(failure)
```

Comment:
If you have a large number of shards, this line can end up massive. I think just logging the first failure is a reasonable tradeoff for visibility

Reply:
@tommyzli What do you think about a reasonable limit (5 failures for example)? The exception would look like the example below:
line1
line2
line3
line4
line5
...and 23 more omitted failures

Reply:
@bartier I like that!